src/CalcViewModel/GraphingCalculator/GraphingSettingsViewModel.h

```diff
@@ -0,0 +1,297 @@
+                        if (value->Length() == sz)
+                        {
+                            m_Graph->YAxisMin = m_YMinValue = number;
+                            YMinError = false;
+                        }
+                        else
+                        {
+                            YMinError = true;
+                        }
+                    }
+                    catch (...)
+                    {
+                        YMinError = true;
+                    }
+                }
+                RaisePropertyChanged("YError");
+                RaisePropertyChanged("YMin");
+                UpdateDisplayRange(false);
+            }
+        }
+
+        property Platform::String ^ YMax
+        {
+            Platform::String ^ get()
+            {
+                return m_YMax;
+            }
+            void set(Platform::String ^  value)
+            {
+                if (m_YMax == value)
+                {
+                    return;
+                }
+                m_YMax = value;
+                m_YIsMinLastChanged = false;
+                if (m_Graph != nullptr)
+                {
+                    try
+                    {
+                        size_t sz;
+                        auto number = std::stod(value->Data(), &sz);
+                        if (value->Length() == sz)
+                        {
+                            m_Graph->YAxisMax = m_YMaxValue = number;
+                            YMaxError = false;
+                        }
+                        else
+                        {
+                            YMaxError = true;
+                        }
+                    }
+                    catch (...)
+                    {
+                        YMaxError = true;
+                    }
+                }
+                RaisePropertyChanged("YError");
+                RaisePropertyChanged("YMax");
+                UpdateDisplayRange(false);
+            }
+        }
+
+        property int TrigUnit
+        {
+            int get()
+            {
+                return m_Graph == nullptr ? (int)Graphing::EvalTrigUnitMode::Invalid : m_Graph->TrigUnitMode;
+            }
+            void set(int value)
+            {
+                if (m_Graph == nullptr)
+                {
+                    return;
+                }
+                m_Graph->TrigUnitMode = value;
+                RaisePropertyChanged(L"TrigUnit");
+            }
+        }
+
+        property bool TrigModeRadians
+        {
+            bool get()
+            {
+                return m_Graph != nullptr && m_Graph->TrigUnitMode == (int)Graphing::EvalTrigUnitMode::Radians;
+            }
+            void set(bool value)
+            {
+                if (value && m_Graph != nullptr && m_Graph->TrigUnitMode != (int)Graphing::EvalTrigUnitMode::Radians)
+                {
+                    m_Graph->TrigUnitMode = (int)Graphing::EvalTrigUnitMode::Radians;
```

Comment:
This is "expected". The Graph control will automatically reset X and Y ranges when you select a different trigo unit, so we call `RefreshPosition()` in the view model after we set a different unit.

Another solution is to not call RefreshPosition and update the UI with the new XMin, Xmax, YMin and YMax values, but it will be very frustrating for users to set these values again.

Reply:
Could we set the correct x/y min/max values after the graph is initialized but before we render the graph to avoid the flickering?